### DISCUSSION TITLE
ANN: Compute Servers -- CoCalc finally has GPU's and very powerful and affordable VM's

### DISCUSSION THREAD

**Discussion by williamstein** [sagemathinc/cocalc#7048] [Announcements]
CoCalc now features robust compute servers, enabling users to connect a remote computer to CoCalc and utilize it for terminals and Jupyter notebooks. These compute servers open up possibilities for enhanced computing resources, extending far beyond the bounds of local machines. Users simply create a compute server in a project, select the software image and (optional) GPU they require, and can then start running any terminal or Jupyter notebook on this server for an on-demand fee, charged by the second when the server is in use.

The GPU support is extensive, offering variants including A100 80GB, A100 40GB, L4, and T4 GPUs with finely configured software stacks. These stack images include SageMath, Google Colab, Julia, PyTorch, Tensorflow and CUDA Toolkit, accommodating a versatile range of uses. The compute servers integrating these GPUs come at highly competitive pricing, particularly for spot instances. CoCalc's compute servers represent a massive enhancement to default projects, offering increased speed, flexibility, and computational power, transforming the way users can utilize CoCalc for their projects.

<img width="903" alt="image" src="https://github.com/sagemathinc/cocalc/assets/1276278/02e86826-8127-4030-bb13-7611526763dd">

To set up a compute server in CoCalc, log into your project, create a compute server through the "Servers" button, selecting your desired software image and optionally a GPU. To use the server, create a terminal file or a Jupyter notebook, move it to the server through the upper left menu, and remember to sync files for editing during computations.

Finally, here is a quick tutorial on how to get started with compute servers on CoCalc:

1. Once logged in, navigate to your project where you intend to use the compute server.
2. Click on the "Servers" button on the left side of the screen and select "Create Compute Server".
3. You will be prompted to select the desired software image and optionally a GPU.  A GPU is selected by default but you can disable it if you don't need one.  If you are going to write code using CUDA libraries, choose the "Cuda Toolkit" image. If you want to accelerate PyTorch computations with a GPU, choose the "PyTorch" image.  If you want to use SageMath, choose the Sage image.
4. Start your compute server.
5. If you want to use the Linux command line, e.g., compilers, etc., create a terminal file (one ending in .term) and using the upper-left menu, select your compute server. If you chose the 'Cuda Toolkit', then the 'nvcc' command will be available for compiling .cu code.
7. If you need to edit the files during your computations on the compute server, remember to click the 'Sync' button at the top left of the terminal for the files to get copied to your compute server.
8. If you chose the "PyTorch" image or similar, create a Jupyter notebook and move it to the compute server via the upper-left menu.  You can then select a Jupyter kernel that's available on the compute server, and your Jupyter notebook will run there.

Remember, compute servers are billed by the second only when they exist.

<img width="811" alt="image" src="https://github.com/sagemathinc/cocalc/assets/1276278/969ab39d-caa4-4682-8552-146306b29ecd">

**Comment by ktaletsk** [comment]
Congrats!
